src/openssl_wrap.cpp

```diff
@@ -140,6 +140,22 @@ SSL_X509_REQ_Ptr _X509_REQ_new()
     return SSL_X509_REQ_Ptr{OpensslCallPtr::callChecked(lib::OpenSSLLib::SSL_X509_REQ_new)};
 }
 
+int _EVP_DigestFinal_ex(EVP_MD_CTX* ctx, unsigned char* md, unsigned int* s) {
```

Comment:
This is the place where OpenSSL error handling should take place. In particular, conversion of integer based function return codes to C++ exceptions. This is what all the ::callChecked API is for. The functions itself should be void if they have no apparent output.

Reply:
done